components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/PostAuthAssociationHandler.java

```diff
@@ -0,0 +1,227 @@
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This PostAuthentication Handler is responsible for handling the association of user accounts with local users.
+ */
+public class PostAuthAssociationHandler extends AbstractPostAuthnHandler {
+
+    private static final Log log = LogFactory.getLog(PostAuthAssociationHandler.class);
+    private static PostAuthAssociationHandler instance;
+    private static final String USER_TENANT_DOMAIN = "user-tenant-domain";
+
+    /**
+     * To get an instance of {@link PostAuthAssociationHandler}.
+     *
+     * @return instance of PostAuthAssociationHandler
+     */
+    public static PostAuthAssociationHandler getInstance() {
+
+        if (instance == null) {
+            synchronized (PostAuthAssociationHandler.class) {
+                if (instance == null) {
+                    instance = new PostAuthAssociationHandler();
+                }
+            }
+        }
+        return instance;
+    }
+
+    @Override
+    public int getPriority() {
+        /* Priority should be greater than PostJitProvisioningHandler, so that JIT provisioned users local claims would
+         passed to the service provider given the assert local mapped user option is selected */
+        return 21;
+    }
+
+    @Override
+    public String getName() {
+
+        return "PostAuthAssociationHandler";
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public PostAuthnHandlerFlowStatus handle(HttpServletRequest request, HttpServletResponse response,
+            AuthenticationContext context) throws PostAuthenticationFailedException {
+
+        SequenceConfig sequenceConfig = context.getSequenceConfig();
+        Map<ClaimMapping, String> authenticatedUserAttributes = null;
+
+        for (Map.Entry<Integer, StepConfig> entry : sequenceConfig.getStepMap().entrySet()) {
+            StepConfig stepConfig = entry.getValue();
+            AuthenticatorConfig authenticatorConfig = stepConfig.getAuthenticatedAutenticator();
+            ApplicationAuthenticator authenticator = authenticatorConfig.getApplicationAuthenticator();
+
+            if (authenticator instanceof FederatedApplicationAuthenticator) {
+                Map<String, String> localClaimValues;
+
+                String originalExternalIdpSubjectValueForThisStep = stepConfig.getAuthenticatedUser()
+                        .getAuthenticatedSubjectIdentifier();
+
+                if (stepConfig.isSubjectIdentifierStep()) {
+                    // there can be only step for subject attributes.
+                    String associatedID = null;
+
+                    // now we know the value of the subject - from the external identity provider.
+                    if (sequenceConfig.getApplicationConfig().isAlwaysSendMappedLocalSubjectId()) {
+
+                        // okay - now we need to find out the corresponding mapped local subject
```

Comment:
Could you re evaluate the comment here? Maybe remove 'ok' and make it a single line.